components/string.rst

```diff
@@ -0,0 +1,378 @@
+:class:`Symfony\\Component\\String\\UnicodeString`, pass the string contents as
+their arguments and then use the object-oriented API to work with those strings::
+
+    use Symfony\Component\String\UnicodeString;
+
+    $text = (new UnicodeString('This is a déjà-vu situation.'))
+        ->trimEnd('.')
+        ->replace('déjà-vu', 'jamais-vu')
+        ->append('!');
+    // $text = 'This is a jamais-vu situation!'
+
+    $content = new UnicodeString('नमस्ते दुनिया');
+    if ($content->lower()->startsWith('नमस्ते')) {
+        // ...
+    }
+
+Method Reference
+----------------
+
+Methods to Create String Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First, you can create objects prepared to store strings as bytes, code points
+and grapheme clusters with the following classes::
+
+    use Symfony\Component\String\ByteString;
+    use Symfony\Component\String\CodePointString;
+    use Symfony\Component\String\UnicodeString;
+
+    $foo = new ByteString('hello');
+    $bar = new CodePointString('hello');
+    // UnicodeString is the most commonly used class
+    $baz = new UnicodeString('hello');
+
+Use the ``wrap()`` static method to instantiate more than one string object::
+
+    $contents = ByteString::wrap(['hello', 'world']);        // $contents = ByteString[]
+    $contents = UnicodeString::wrap(['I', '❤️', 'Symfony']); // $contents = UnicodeString[]
+
+    // use the unwrap method to make the inverse conversion
+    $contents = UnicodeString::unwrap([
+        new UnicodeString('hello'), new UnicodeString('world'),
+    ]); // $contents = ['hello', 'world']
+
+There are two shortcut functions called ``b()`` and ``u()`` to create
+``ByteString`` and ``UnicodeString`` objects::
+
+    // ...
+    use function Symfony\Component\String\b;
+    use function Symfony\Component\String\u;
+
+    // both are equivalent
+    $foo = new ByteString('hello');
+    $foo = b('hello');
+
+    // both are equivalent
+    $baz = new UnicodeString('hello');
+    $baz = u('hello');
+
+There are also some specialized constructors::
+
+    // ByteString can create a random string of the given length
+    $foo = ByteString::fromRandom(12);
+
+    // CodePointString and UnicodeString can create a string from code points
+    $foo = UnicodeString::fromCodePoints(2344, 2350, 2360, 2381, 2340, 2375);
+    // $foo = 'नमस्ते'
+
+Methods to Transform String Objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each string object can be transformed into the other two types of objects::
+
+    $foo = ByteString::fromRandom(12)->toCodePointString();
+    $foo = (new CodePointString('hello'))->toUnicodeString();
+    $foo = UnicodeString::fromCodePoints(104, 101, 108, 108, 111)->toByteString();
+
+Methods Related to Length and White Spaces
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+::
+
+    // returns the number of graphemes, code points or bytes of the given string
+    (new ByteString('नमस्ते'))->length();      // 18 (bytes)
```

Comment:
maybe put `नमस्ते` is a variable to make it crystal clear that this is the same value everywhere?